assets/src/edit-story/components/inspector/inspectorProvider.js

```diff
@@ -6,26 +6,78 @@ import PropTypes from 'prop-types';
 /**
  * WordPress dependencies
  */
-import { useState } from '@wordpress/element';
+import { useCallback, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
+import { useAPI } from '../../app/api';
 import Context from './context';
 
 const DESIGN = 'design';
 const DOCUMENT = 'document';
 const PREPUBLISH = 'prepublish';
 
 function InspectorProvider( { children } ) {
+	const { actions: { getAllStatuses, getAllUsers } } = useAPI();
 	const [ tab, setTab ] = useState( DESIGN );
+	const [ users, setUsers ] = useState( [] );
+	const [ statuses, setStatuses ] = useState( [] );
+
+	const [ isUsersLoading, setIsUsersLoading ] = useState( false );
+	const [ isStatusesLoading, setIsStatusesLoading ] = useState( false );
+	const [ disabledStatuses, setDisabledStatuses ] = useState( [] );
+
+	const loadStatuses = useCallback( () => {
+		if ( ! isStatusesLoading && statuses.length === 0 ) {
```

Comment:
I'd much rather not call this callback if there already are statuses rather than check every time.